src/nni_manager/training_service/local/localTrainingService.ts

```diff
@@ -184,6 +184,18 @@ class LocalTrainingService implements TrainingService {
         return trialJob;
     }
 
+    public async getTrialLog(trialJobId: string, logType: LogType): Promise<string> {
+        let logPath: string;
+        if (logType === 'TRIAL_LOG') {
+            logPath = path.join(this.rootDir, 'trials', trialJobId, 'trial.log');
+        } else if (logType === 'TRIAL_STDERR') {
+            logPath = path.join(this.rootDir, 'trials', trialJobId, 'stderr');
+        } else {
+            throw new Error('unexpected log type');
+        }
+        return fs.promises.readFile(logPath, 'utf8');
```

Comment:
does this readFile() function need to catch exception?

Reply:
This exception should be catched by `getTrialLog` in `restHandler.ts`